cmsdb/campaigns/run3_2022_postEE_nano_v11/qcd.py

```diff
@@ -0,0 +1,397 @@
+# coding: utf-8
+
+"""
+QCD datasets for the 2022 post-EE data-taking campaign
+"""
+
+import cmsdb.processes as procs
+from cmsdb.campaigns.run3_2022_postEE_nano_v11 import campaign_run3_2022_postEE_nano_v11 as cpn
+
+
+#
+# QCD flat samples
+#
+
+# GrASP: https://cms-pdmv.cern.ch/grasp/samples?campaign=Run3Summer22EEGS&nanoaod_version=v11&dataset=QCD_PT-15*Flat2018&chained_request=Premix
+# missing as of 2023-07-01
+# cpn.add_dataset(
+#     name="qcd_flat2018_pythia",
+#     id=None,
+#     is_data=False,
+#     processes=[procs.qcd_flat],
+#     keys=[
+#         "/QCD_PT-15_TuneCP5_Flat2018_13p6TeV_pythia8/Run3Summer22EENanoAODv11-126X_mcRun3_2022_realistic_postEE_v1-v2/NANOAODSIM",
+#     ],
+#     n_files=None,
+#     n_events=None,
+# )
+
+# GrASP: https://cms-pdmv.cern.ch/grasp/samples?campaign=Run3Summer22EEGS&nanoaod_version=v11&dataset=QCD_PT-15*Flat2018&chained_request=EpsilonPU
+cpn.add_dataset(
+    name="qcd_flat2018_epsilonPU_pythia",
```

Comment:
Just out of curiosity, what does flat2018 mean?

Reply:
AFAIK the 'flat' part has to do with how the events are sampled. Since QCD is a steeply falling spectrum, sampling directly according to the cross section would lead mostly to low-pT events and a sparsely populated high-pT region. To counteract this, the sampled spectrum is biased by an exponential function that approximates the falling cross section, so that events are sampled evenly across pT, i.e. the unweighted events have an approximately flat distribution. To restore the steeply falling spectrum, the inverse of the bias function is applied as an event weight. There's some more information on this in the [Pythia manual](https://pythia.org/latest-manual/PhaseSpaceCuts.html) if you are interested.

As for the '2018' part, I have no idea :) Maybe it refers to the fact that the bias parameters were taken from the 2018 campaign?

Reply:
Ha, didn't know. Thanks!